api/clients/accountant.go

```diff
@@ -160,6 +159,59 @@ func (a *accountant) GetRelativeBinRecord(index uint32) *BinRecord {
 	return &a.binRecords[relativeIndex]
 }
 
+func (a *accountant) SetPaymentState(paymentState *disperser_rpc.GetPaymentStateReply) error {
```

Comment:
This feels too coupled to me. A Public setter that is not part of the interface and it set by some response from another client. As discussed in other threads, could we not combine accountant and disperser_client more smoothly? Does accountant really need to be an independent struct?

Reply:
We wanted to keep accounting flow separate from the client, but arguably we can combine them into a bigger struct. What do you think @ian-shim ?

Reply:
Makes sense! I would also not try to future-proof this design too much (to a case where there are multiple dispersers and you need some super complex accountant that manages multiple dispersers). We should try to make the simple case simple, true to golang's philosophy, and people who need complex integrations can take on the complexity hit around our simple API.

Another possibility is to have 2 constructors: one where we inject the accountant, and the other that initializes the accountant by calling the disperser?

Reply:
Hmm all one needs to do is to call the current constructor and then call `PopulateAccountant`, so adding a separate constructor doesn't seem that beneficial to me. I won't mind adding it, probably in the follow-up [PR](https://github.com/Layr-Labs/eigenda/pull/965) that focus on integrations adaptions (maybe something like `NewDisperserClientV2AutoPay`?